api-server/src/main/kotlin/com/unicorn/api/controller/medicine/MedicineController.kt

```diff
@@ -0,0 +1,32 @@
+package com.unicorn.api.controller.medicine
+
+import com.unicorn.api.controller.api_response.ResponseError
+import com.unicorn.api.controller.user.UserPostRequest
+import com.unicorn.api.domain.account.UID
+import com.unicorn.api.query_service.medicine.MedicineQueryService
+import com.unicorn.api.query_service.medicine.MedicineResult
+import com.unicorn.api.query_service.user.UserQueryService
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+@RestController
+class MedicineController(
+    private val medicineQueryService: MedicineQueryService,
+    private val userQueryService: UserQueryService
+) {
+    @GetMapping("/medicines")
+    fun getMedicines(@RequestHeader("X-UID") uid: String): ResponseEntity<*> {
+        return try {
+            val user = userQueryService.getOrNullBy(uid)
+
+            if (user != null) {
+                val result = medicineQueryService.getMedicines(uid)
+                ResponseEntity.ok(result)
+            } else {
+                ResponseEntity.status(400).body("User not found")
```

Comment:
[super-nits]
ここは自分が用意したRequestErrorを用いると erroTypeプロパティにmessageを入れることができるので、もしよければ！！！
```suggestion
                ResponseEntity.status(400).body(ResponseError("User not found"))
```